providers/VercelChatProvider.tsx

```diff
@@ -53,7 +73,19 @@ export function VercelChatProvider({
     setMessages,
     reload,
     append,
-  } = useVercelChat({ id: chatId, initialMessages });
+  } = useVercelChat({ 
+    id: chatId, 
+    initialMessages,
+    uploadedAttachments // Pass attachments to useVercelChat
+  });
+  
+  // When a message is sent successfully, clear the attachments
+  const handleSendMessageWithClear = async (event: React.FormEvent<HTMLFormElement>) => {
+    await handleSendMessage(event);
+    
+    // Clear attachments after sending
+    clearAttachments();
+  };
```

Comment:
why not just call this in `onFinish` callback in `useChat` hook?

Reply:
Nope, 
it will keep showing files above the input even after the user submits the message, and it will be showing images even when the AI message is streaming. Suppose AI takes 30 seconds to answer (reasoning + response), the user will still be looking at the files above the input and it will only disappear when message stream end. 

It will be a bad UX and we just attach files with user's message and submit. then we don't have any use of files after the submission.